hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServerResponseEncoder.java

```diff
@@ -55,4 +69,33 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
       ctx.write(msg, promise);
     }
   }
+
+  private boolean handleFatalThreshold(ChannelHandlerContext ctx) {
+    int fatalThreshold = rpcServer.getWriteBufferFatalThreshold();
```

Comment:
We just need this threshold? Let's just pass the threshold in?

Reply:
Not sure what you mean here. You mean instead of exposing a getter?

I want to use a getter here so that the fatal threshold can be live updated with `update_config`. If I passed it into the NettyRpcServerResponseEncoder, it would be static for the lifetime of a connection. I could pass in an IntSupplier instead, but is that much better than passing in the NettyRpcServer? I'm happy to do that if preferred

Reply:
Let's pass an IntSupplier and add comment to say we want to support update_config so we can not pass the threshold in directly.